plugins/auto-sizes/tests/test-improve-calculate-sizes.php

```diff
@@ -646,6 +646,565 @@ public function data_image_blocks_with_relative_alignment_for_classic_theme(): a
+			'Return 66.66% width of wideSize 853px, parent block wide alignment, image block right alignment'      => array(
+				'wide',
+				'right',
+				'sizes="(max-width: 853px) 100vw, 853px" ',
+			),
+
+			// Parent full alignment.
+			'Return 66.66% width of contentSize 413px, parent block full alignment, image block default alignment'    => array(
+				'full',
+				'',
+				'sizes="(max-width: 413px) 100vw, 413px" ',
+			),
+			'Return 66.66% width of wideSize 853px, parent block full alignment, image block wide alignment'         => array(
+				'full',
+				'wide',
+				'sizes="(max-width: 853px) 100vw, 853px" ',
+			),
+			'Return full size, parent block full alignment, image block full alignment'               => array(
+				'full',
+				'full',
+				'sizes="100vw" ',
+			),
+			'Return 66.66% width of wideSize 853px, parent block full alignment, image block left alignment'       => array(
+				'full',
+				'left',
+				'sizes="(max-width: 853px) 100vw, 853px" ',
+			),
+			'Return 66.66% width of contentSize 413px, parent block full alignment, image block center alignment'     => array(
+				'full',
+				'center',
+				'sizes="(max-width: 413px) 100vw, 413px" ',
+			),
+			'Return 66.66% width of wideSize 853px, parent block full alignment, image block right alignment'      => array(
+				'full',
+				'right',
+				'sizes="(max-width: 853px) 100vw, 853px" ',
+			),
+		);
+	}
+
+	/**
+	 * Test that the layout property of a column block is passed by context to the image block.
+	 *
+	 * @dataProvider data_image_block_with_parent_columns_and_its_parent_group_block
+	 *
+	 * @param string $group_block_alignment   Group block alignment.
+	 * @param string $columns_block_alignment Columns block alignment.
+	 * @param string $image_block_alignment   Image block alignment.
+	 * @param string $expected                Expected output.
+	 */
+	public function test_image_block_with_parent_columns_and_its_parent_group_block( string $group_block_alignment, string $columns_block_alignment, string $image_block_alignment, string $expected ): void {
+		// Skip test for WordPress versions below 6.8.
+		if ( version_compare( get_bloginfo( 'version' ), '6.8', '<' ) ) {
+			$this->markTestSkipped( 'This test requires WordPress 6.8 or higher.' );
+		}
+
+		$block_content = $this->get_group_block_markup(
+			$this->get_columns_block_markup(
+				$this->get_image_block_markup( self::$image_id, 'large', $image_block_alignment ),
+				array(
+					'align' => $columns_block_alignment,
+				),
+				array(
+					'66.66%' => true,
+					'33.33%' => false,
+				)
+			),
+			array(
+				'align' => $group_block_alignment,
+			)
+		);
+
+		$result = apply_filters( 'the_content', $block_content );
+
+		$this->assertStringContainsString( $expected, $result );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array<string, array<int, string>> The ancestor and image alignments.
+	 */
+	public function data_image_block_with_parent_columns_and_its_parent_group_block(): array {
```

Comment:
Each of the tests has the exact same sizes attribute: `sizes="(max-width: 413px) 100vw, 413px"`.

Is this expected?

Reply:
Yes, this is expected as it end up to the `default` alignment any of it's parents or image alignment as `default` alignment. In https://github.com/WordPress/performance/pull/1818/commits/bba49979a92a72a50d50db94262109239656d456, I added more variations of alignment options.